SCClassLibrary/SCDoc/SCDocRenderer.sc

```diff
@@ -51,6 +51,111 @@ SCDocHTMLRenderer {
 		^str.replace(" ", "%20")
 	}
 
+	*replaceRegexp { |source, findRegexp, replace|
```

Comment:
I don't think you need this. Why use regexp? You can use all of MathJax's syntax by simply using an idiom markup from SCDoc, and replacing it with `$...$` for inline, or `$$....$$` for displayed formulas. That's it.

You can even write arrays like this:


```
math::
\begin{array}{c|lcr}
n & \text{Left} & \text{Center} & \text{Right} \\
\hline
1 & 0.24 & 1 & 125 \\
2 & -1 & 189 & -8 \\
3 & -20 & 2000 & 1+10i
\end{array}
::

```

Will render as:

$$
\begin{array}{c|lcr}
n & \text{Left} & \text{Center} & \text{Right} \\
\hline
1 & 0.24 & 1 & 125 \\
2 & -1 & 189 & -8 \\
3 & -20 & 2000 & 1+10i
\end{array}
$$

Reply:
And so on. Keep it simple.

Reply:
Try that and let's see what happens.


![image](https://github.com/supercollider/supercollider/assets/271230/b8bf02ca-0b09-41f6-b9a9-b303335aa44b)

Reply:
I hope I made myself clear. If not, just reach out. Don't stress about it.


https://github.com/supercollider/supercollider/assets/271230/d9a284e3-e7cf-48f3-ba29-b339652df3a5

Reply:
- `regexp` is only needed to implement MathML.  Attributes can follow or precede tags, and typing all possible cases is a horrible thing to do.
- I think to enable the use of `math:: ... ::` I should change some parts of C++ in the source, but I cannot do it as I do not understand C++.

The details of the answers are in the following post in this thread, since some related things are discussed quite a lot:
 https://github.com/supercollider/supercollider/pull/6260#issuecomment-2079328847

Reply:
Let's forget MathML, it does not fit the purpose of this PR.

Reply:
Done!
https://github.com/supercollider/supercollider/commit/be2de2c46486618325e5235b437459284a420c5a